CIP-????/README.md

```diff
@@ -0,0 +1,205 @@
+---
+CIP: ????
+Title: Representing the Script Context as a Sums-of-Products Term
+Authors: 
+  - Michael Peyton Jones <michael.peyton-jones@iohk.io>
+Implementors: 
+  - Michael Peyton Jones <michael.peyton-jones@iohk.io>
+Status: Proposed
+Category: Plutus
+Created: 2023-04-20
+Discussions: 
+  - https://github.com/cardano-foundation/CIPs/pull/455
+License: CC-BY-4.0
+---
+
+## Abstract
+
+Plutus scripts receive information about the transaction currently being validated by being passed an argument containing that information, called the "script context".
+The script context is currently represented in Plutus Core using the `Data` builtin type. 
+This is slow to process, so we propose to use the support for sums-of-products (SOPs) in Plutus Core (proposed in CIP-85) to represent the script context directly as a term using sums-of-products.
+
+## Motivation: why is this CIP necessary?
+
+## Background
+
+This CIP assumes familiarity with the concepts described in CIP-85, including SOPs and the origin of `Data`.
+
+## The Problem
+
+The problem is:
+1. It is slow to perform any kind of manipulation of the script context (see CIP-85 for benchmarking of operations on `Data` versus SOPs)
+2. Translating ("decoding") the script context into a native form ([Scott encoding][] or SOPs) is far too slow because of 1 and because script contexts become large
+
+These combine to make decoding real world script contexts prohibitive in many cases:
+budget profiling of examples in `plutus-use-cases` that naively decode the script context revealed that many were spending 30-40% of their budget just on the decoding step, which is unacceptable.
```

Comment:
Given that only PlutusTx (as far as I know?) is trying to translate the script context into some intermediate representation, I'd assume that yes. 

So it could be interesting to get some before/after benchmarks for solutions like Plu-ts or Aiken regarding the impact of using SOP vs Data. Though, if the script context still comes in the form of `Data` from the ledger, I believe that the overhead of turning a `Data` into a SOP will be still high.

Reply:
Yes, this is really only for people who actually try to convert the script context into another form. Although I do still think this is _desirable_ - it's just obviously too expensive to be worth it at the moment. 

So the numbers here are more meant to indicate how unacceptably high the cost is, _not_ to argue that these are reductions that real users might see (the test cases in question are _not_ real). I can see that's misleading, I'll rephrase.